src/Twig/FilesystemLoader.php

```diff
@@ -205,4 +205,56 @@ function (DirectoryInterface $dir) {
 
         throw new LoaderError($this->errorCache[$name]);
     }
+
+    /**
+     * {@inheritdoc}
+     */
+    private function normalizeName($name)
+    {
+        return preg_replace('#/{2,}#', '/', str_replace('\\', '/', $name));
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    private function parseName($name, $default = self::MAIN_NAMESPACE)
+    {
+        if (isset($name[0]) && '@' == $name[0]) {
+            if (false === $pos = strpos($name, '/')) {
+                throw new LoaderError(sprintf('Malformed namespaced template name "%s" (expecting "@namespace/template_name").', $name));
+            }
+
+            $namespace = substr($name, 1, $pos - 1);
+            $shortname = substr($name, $pos + 1);
+
+            return array($namespace, $shortname);
+        }
+
+        return array($default, $name);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    private function validateName($name)
+    {
+        if (false !== strpos($name, "\0")) {
+            throw new LoaderError('A template name cannot contain NUL bytes.');
+        }
+
+        $name = ltrim($name, '/');
+        $parts = explode('/', $name);
+        $level = 0;
+        foreach ($parts as $part) {
+            if ('..' === $part) {
+                --$level;
+            } elseif ('.' !== $part) {
+                ++$level;
+            }
+
+            if ($level < 0) {
+                throw new LoaderError(sprintf('Looks like you try to load a template outside configured directories (%s).', $name));
+            }
+        }
+    }
```

Comment:
😢 Is there any need to extend Symfony's Filesystem Loader at this point? Probably not. 

I think we could do `validateName` differently since our filesystem does some path normalization and root violation check. Either skip it or call our normalize path or something...